### DISCUSSION TITLE
CORS Issue with Traefik / Cloudflare

### DISCUSSION THREAD

**Discussion by timbmg** [plausible/analytics#2137] [Self-Hosted Support]
I am running plausible instance at analytics.example.com. I configured it with traefik, and can access the plausible. I have set up a Cloudflare worker [according to the docs](https://plausible.io/docs/proxy/guides/cloudflare). And added the script to my website like this:
```html
<script defer
  data-domain="example.com"
  data-api="https://my-cloudflare-worker.workers.dev/my-folder/event"
  src="https://my-cloudflare-worker.workers.dev/my-folder/script.js"
</script>
```
When going to example.com, I see the CORS error in the console:
```
Access to XMLHttoRequest at 'https:/my-cloudflare-worker.workers.dev/my-folder/event' from origin 'https:/example.com' has
been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```
I checked the script endpoint and its working (ie it returns the script when i just put the url in the browser).

When I dont proxy the script via cloudflare, it seems to be working.

Any clue what might be missing?

**Comment by samhibberd** [comment]
Did you ever find a solution to this, we are running into the same issue?

**Comment by ruslandoga** [comment]
👋 @timbmg @samhibberd 

I've just went through the guide to setup cf workers proxy on https://stats.edify.space and it seems to be working. I wonder if you could share more info on your setup, what worker script you are using, what your `<script>` tag looks like, etc.

<details><summary>Here're mine:</summary>

```js
const ScriptName = '/js/script.js';
const Endpoint = '/api/event';

const ScriptWithoutExtension = ScriptName.replace('.js', '')

addEventListener('fetch', event => {
    event.passThroughOnException();
    event.respondWith(handleRequest(event));
})

async function handleRequest(event) {
  const pathname = new URL(event.request.url).pathname
  const [baseUri, ...extensions] = pathname.split('.')

  if (baseUri.endsWith(ScriptWithoutExtension)) {
      return getScript(event, extensions)
  } else if (pathname.endsWith(Endpoint)) {
      return postData(event)
  }
  return new Response(null, { status: 404 })
}

async function getScript(event, extensions) {
    let response = await caches.default.match(event.request);
    if (!response) {
        response = await fetch("https://who.copycat.fun/js/plausible." + extensions.join("."));
        event.waitUntil(caches.default.put(event.request, response.clone()));
    }
    return response;
}

async function postData(event) {
    const request = new Request(event.request);
    request.headers.delete('cookie');
    return await fetch("https://who.copycat.fun/api/event", request);
}
```

```html
<script
  defer
  data-domain="stats.copycat.fun"
  data-api="https://hello-world-rough-snow-8d8e.test1086.workers.dev/api/event"
  src="https://hello-world-rough-snow-8d8e.test1086.workers.dev/js/script.js"
>
</script>
```

</details>

**Reply by ruslandoga** [reply]
Notice that the default script uses `/api/event` suffix for the `event` check, so I'm using `https://hello-world-rough-snow-8d8e.test1086.workers.dev/api/event` in the `<script data-api>` attribute.

**Reply by samhibberd** [reply]
Funnily enough this was exactly it, was missing the `event/api` suffix off the data-api attribute. Adding that resolved. Maybe docs not 100% clear, or most likely me rushing 😉

**Reply by ruslandoga** [reply]
`api/event` is mentioned in https://plausible.io/docs/proxy/guides/cloudflare#step-3-paste-the-following-code, but I missed it too on my first attempt :)
